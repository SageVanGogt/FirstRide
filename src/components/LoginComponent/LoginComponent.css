.login-container {
  position: fixed;
  z-index: 3;
  display: flex;
  align-items: center;
  background-color: rgba(249,194,85, 0.7);
  left: calc(50% - 350px);
  right: calc(50% - 350px);
  top: calc(50% - 200px);
  bottom: calc(50% - 200px);
  height: 400px;
  width: 700px;
  border-radius: 8px;
}

.back-btn {
  position: absolute;
  top: 5px;
  right: 5px;
  padding: 3px 5px;
  background: none;
  color: white;
  border: 2px solid white;
  border-radius: 5px;
  font-family: 'Roboto', sans-serif; 
  z-index: 4;
}

.back-btn:hover {
  border: 2px solid #476399;
  color: #476399;  
}
