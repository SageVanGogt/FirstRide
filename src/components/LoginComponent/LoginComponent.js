import React from 'react';
import PropTypes from 'prop-types';
import SigninContainer from 
  './../../containers/SigninContainer/SigninContainer';
import SignupContainer from 
  './../../containers/SignupContainer/SignupContainer';
import './LoginComponent.css';

const LoginComponent = ({toggleLogin}) => (
  <div className="login-backdrop">
    <div className="login-container">
      <div 
        className="back-btn"
        onClick={toggleLogin}>
          X
      </div>
      <SigninContainer />
      <SignupContainer />
    </div> 
  </div>
);


LoginComponent.propTypes = {
  toggleLogin: PropTypes.func
};

export default LoginComponent;