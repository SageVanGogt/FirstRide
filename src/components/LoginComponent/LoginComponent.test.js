import React from 'react';
import { shallow } from 'enzyme';
import LoginComponent from './LoginComponent';

describe('LoginComponent', () => {
  it('should match the snapshot', () => {
    let wrapper = shallow(<LoginComponent />)

    expect(wrapper).toMatchSnapshot();
  })
})