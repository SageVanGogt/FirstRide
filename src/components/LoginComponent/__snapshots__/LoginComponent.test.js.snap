// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="login-container"
>
        <div
                className="back-btn"
                onClick={undefined}
        >
                back
        </div>
        <Connect(SigninContainer) />
        <Connect(SignupContainer) />
</div>,
      "className": "login-backdrop",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="back-btn"
            onClick={undefined}
>
            back
</div>,
          <Connect(SigninContainer) />,
          <Connect(SignupContainer) />,
        ],
        "className": "login-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "back",
            "className": "back-btn",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "back",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="login-container"
>
          <div
                    className="back-btn"
                    onClick={undefined}
          >
                    back
          </div>
          <Connect(SigninContainer) />
          <Connect(SignupContainer) />
</div>,
        "className": "login-backdrop",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="back-btn"
              onClick={undefined}
>
              back
</div>,
            <Connect(SigninContainer) />,
            <Connect(SignupContainer) />,
          ],
          "className": "login-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "back",
              "className": "back-btn",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "back",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
